### PR TITLE
chore(lib/common): remove Equal method for Hash and replace with ==

### LINKED ISSUE #2939: `lib/common`: remove `Equal` method for `Hash` and replace with `==`
We can equal hashes (32 bytes arrays) using just `==`, there is no point to have an `Equal` method, and even less to check equality using the method function body which converts both arrays to bytes slices and then to strings to check for equality

### PR DESCRIPTION
**Pull request by lakshya-sky**
## Issues

Fixes #2939

### CODE REVIEW

File: dot/state/block.go [hunk 1]
```diff
@@ -621,7 +621,7 @@ func (bs *BlockState) HandleRuntimeChanges(newState *rtstorage.TrieState,
 
 	codeSubBlockHash := bs.baseState.LoadCodeSubstitutedBlockHash()
 
-	if !codeSubBlockHash.Equal(common.Hash{}) {
+	if codeSubBlockHash != common.EmptyHash() {
```
Comment (qdm12) [inline dot/state/block.go:624]: nit keep it to

```suggestion
	if codeSubBlockHash != common.Hash{} {
```

not sure why we have `EmptyHash` in the first place, might be worth removing it as well.

File: lib/common/hash.go [hunk 2]
```diff
@@ -22,6 +22,13 @@ const (
 // Hash used to store a blake2b hash
 type Hash [32]byte
 
+// EmptyHash returns Hash which is an array of 32bytes with
+// every byte set to zero.
+func EmptyHash() Hash {
+	res := Hash{}
+	return res
+}
```
Comment (qdm12) [inline lib/common/hash.go:30]: Yeah we don't want that, there is no point for it and performance-wise it adds a function call.

**Review by qdm12 — CHANGES_REQUESTED** [review]
Thanks! Just need to remove that `EmptyHash` function :wink:

File: dot/state/block.go (hunk 1 above)
Comment (lakshya-sky) [reply dot/state/block.go:624]:
```
dot/state/block.go:624:39: expected ';', found '{' (typecheck)
        if codeSubBlockHash != common.Hash{} {
```

Comment (lakshya-sky) [reply dot/state/block.go:624]: if I use 
```
codeSubBlockHash != (common.Hash{})
``` 
then error goes away, golang is so strange.

File: lib/common/hash.go [hunk 3]
```diff
@@ -41,7 +41,7 @@ func HashValidator(field reflect.Value) interface{} {
 	// Try to convert to hash type.
 	if valuer, ok := field.Interface().(Hash); ok {
 		// Check if the hash is empty.
-		if valuer.Equal(Hash{}) {
+		if valuer == (Hash{}) {
 			return ""
```
Comment (qdm12) [inline lib/common/hash.go:45]: Let's remove `func (h Hash) Equal(g Hash) bool {` below

**Review by qdm12 — CHANGES_REQUESTED** [review]

**Review by qdm12 — APPROVED** [review]

**Review by EclesioMeloJunior — APPROVED** [review]

**Comment by qdm12** [comment]
Awesome, merged & thanks Darshan :+1:

**Pull request merged**